numpy/lib/tests/test_arraysetops.py

```diff
@@ -32,7 +32,46 @@ def test_intersect1d(self):
         assert_array_equal(c, ed)
 
         assert_array_equal([], intersect1d([], []))
-
+        
+    def test_intersect1d_indices(self):
+        # unique inputs
+        a = np.array([1, 2, 3, 4]) 
+        b = np.array([2, 1, 4, 6])
+        c, i1, i2 = intersect1d(a, b, assume_unique=True, return_indices=True)
+        ee = np.array([1, 2, 4])
+        assert_array_equal(c, ee)
+        assert_array_equal(a[i1], ee)
+        assert_array_equal(b[i2], ee)
+        
+        # non-unique inputs
+        a = np.array([1, 2, 2, 3, 4, 3, 2])
+        b = np.array([1, 8, 4, 2, 2, 3, 2, 3])
+        c, i1, i2 = intersect1d(a, b, return_indices=True)
+        ef = np.array([1, 2, 3, 4])
+        assert_array_equal(c, ef)
+        assert_array_equal(a[i1], ef)
+        assert_array_equal(b[i2], ef)
+                
+        #non1d, unique inputs
+        a = np.array([[2,4,5,6],[7,8,2,15]])
+        b = np.array([[3,2,7,6],[10,12,8,9]])
+        c,i1,i2 = intersect1d(a,b,return_indices=True)
```

Comment:
Did you mean to add `assume_unique` here?

Should be spaces around the commas too